### PR TITLE
Timeslots

### PR DESCRIPTION
**Pull request by KiraBurova**
- [x] Validate overlapping of timeperiods
- [x] Validate that timeperiod belongs to the passed day
- [x] Check that create timeslots and book timeslot are working
- [x] Write tests for two validation functions

### CODE REVIEW

File: timeslots/handler.go [hunk 1]
```diff
@@ -17,23 +21,39 @@ func CreateTimeslots(w http.ResponseWriter, r *http.Request) {
 	var timeslotsData TimeslotData
 	json.NewDecoder(r.Body).Decode(&timeslotsData)
 
+	// check if BookingDay was not send
+	if timeslotsData.BookingDay.IsZero() {
+		w.WriteHeader(http.StatusInternalServerError)
```
Comment (rutaka-n) [inline timeslots/handler.go:26]: If the client does not send `BookingDay` does it mean that the request is bad?

File: timeslots/handler.go [hunk 2]
```diff
@@ -17,23 +21,39 @@ func CreateTimeslots(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
 	repo := NewRepository(db.DbInstance)
 
+	if !timeperiodsBelongToTheDay(timeslotsData.Time, timeslotsData.BookingDay) {
+		w.WriteHeader(http.StatusInternalServerError)
```
Comment (rutaka-n) [inline timeslots/handler.go:33]: Bad Request?

File: timeslots/handler.go [hunk 3]
```diff
@@ -17,23 +21,39 @@ func CreateTimeslots(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	if areTimePeriodsOverlapping(timeslotsData.Time) {
+		w.WriteHeader(http.StatusInternalServerError)
```
Comment (rutaka-n) [inline timeslots/handler.go:38]: Bad Request?

File: timeslots/handler.go [hunk 4]
```diff
@@ -17,23 +21,39 @@ func CreateTimeslots(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
 	for _, timePeriod := range timeslotsData.Time {
 		if !isTimePeriodValid(timePeriod) {
+			w.WriteHeader(http.StatusInternalServerError)
```
Comment (rutaka-n) [inline timeslots/handler.go:44]: Bad Request?

File: timeslots/timeslots_test.go [hunk 5]
```diff
@@ -22,3 +22,57 @@ func TestIsTimePeriodValid(t *testing.T) {
 		assert.False(t, valid)
 	})
 }
+
+func TestAreTimeslotsOverlapping(t *testing.T) {
```
Comment (rutaka-n) [inline timeslots/timeslots_test.go:26]: Tests look good 👍 
I would also add one more test to check the edge case when one period goes right after the previous one.

File: timeslots/timeslots.go [hunk 6]
```diff
@@ -31,23 +33,31 @@ func isTimePeriodValid(timeperiod TimePeriod) bool {
...
+}
 
-	return false
+func timeperiodsBelongToTheDay(timeperiods []TimePeriod, day time.Time) bool {
+	for i := 0; i < len(timeperiods); i++ {
+		period := timeperiods[i]
+		fromBelongsToDay := period.From.Day() == day.Day() && period.From.Month() == day.Month() && period.From.Year() == day.Year()
+		toBelongsToDay := period.To.Day() == day.Day() && period.To.Month() == day.Month() && period.To.Year() == day.Year()
+
+		if !fromBelongsToDay || !toBelongsToDay {
```
Comment (rutaka-n) [inline timeslots/timeslots.go:58]: nit: it might be easier to read the condition if group them, e.g. `!(fromBelongsToDay && toBelongsToDay)`

File: timeslots/repository.go [hunk 7]
```diff
@@ -15,19 +15,32 @@ func NewRepository(db *sql.DB) *Repository {
...
+	// for _, timeslot := range timeslots {
+	// somehow construct SQL query here
+	// }
+
+	timeInUnixFrom := timeslot.TimeFrom.Unix()
+	timeInUnixTo := timeslot.TimeTo.Unix()
+
+	t := TimeslotInDB{TimeslotBase: TimeslotBase{Id: timeslot.Id, OwnerId: timeslot.OwnerId, Booked: false}, TimeFrom: timeInUnixFrom, TimeTo: timeInUnixTo}
+
+	query := "INSERT INTO timeslots(id, ownerId, bookedById, timeFrom, timeTo, booked) values(?,?,?,?,?,?)"
```
Comment (KiraBurova) [inline timeslots/repository.go:41]: remove timeslots on the passted day before insert

**Review by rutaka-n — CHANGES_REQUESTED** [review]
